### ISSUE TITLE
When evaluating LLMs we should stop using cross-entropy against the prompt

### ISSUE THREAD

**Issue by sogartar** [nod-ai/shark-ai#2494]
Right now when [evaluating](https://github.com/nod-ai/shark-ai/blob/0c254262c8c7d0d1643d418a65cc2e9822071a48/sharktank/sharktank/utils/llm_utils.py#L739) we are using the average `ln(softmax(logits))` across all tokens.
The problem with cross-entropy is that when there are unlikely tokens in the prompt sequence then they will dominate the result. They will have low probability predicted by the model. Then their cross-entropy will tend to `-inf` as they approach 0. Then small variations of the predicted probability will yield large changes in the cross-entropy.
E.g.
```
ln(1e-5) = -11.512925464970229
ln(1e-10) = -23.025850929940457
```

From a practical point of view when interpreting the results a probability of `1e-5` is almost the same as `1e-10`. Its like noise. But this difference will dominate the average cross-entropy giving you a large error. We would be largely measuring the noise.
